packages/backend-common/src/reading/AwsCodeCommitUrlReader.ts

```diff
@@ -0,0 +1,393 @@
+    accessKeyId: string,
+    secretAccessKey: string,
+  ): AwsCredentialIdentityProvider {
+    return async () => {
+      return {
+        accessKeyId,
+        secretAccessKey,
+      };
+    };
+  }
+
+  private static async buildCredentials(
+    credsManager: AwsCredentialsManager,
+    region: string,
+    integration?: AwsCodeCommitIntegration,
+  ): Promise<AwsCredentialIdentityProvider> {
+    // Fall back to the default credential chain if neither account ID
+    // nor explicit credentials are provided
+    if (!integration) {
+      return (await credsManager.getCredentialProvider()).sdkCredentialProvider;
+    }
+
+    const accessKeyId = integration.config.accessKeyId;
+    const secretAccessKey = integration.config.secretAccessKey;
+    let explicitCredentials: AwsCredentialIdentityProvider;
+    if (accessKeyId && secretAccessKey) {
+      explicitCredentials = AwsCodeCommitUrlReader.buildStaticCredentials(
+        accessKeyId,
+        secretAccessKey,
+      );
+    } else {
+      explicitCredentials = (await credsManager.getCredentialProvider())
+        .sdkCredentialProvider;
+    }
+
+    const roleArn = integration.config.roleArn;
+    if (roleArn) {
+      return fromTemporaryCredentials({
+        masterCredentials: explicitCredentials,
+        params: {
+          RoleSessionName: 'backstage-aws-code-commit-url-reader',
+          RoleArn: roleArn,
+          ExternalId: integration.config.externalId,
+        },
+        clientConfig: { region },
+      });
+    }
+
+    return explicitCredentials;
+  }
+
+  private async buildCodeCommitClient(
+    credsManager: AwsCredentialsManager,
+    region: string,
+    integration: AwsCodeCommitIntegration,
+  ): Promise<CodeCommitClient> {
+    const credentials = await AwsCodeCommitUrlReader.buildCredentials(
+      credsManager,
+      region,
+      integration,
+    );
+
+    const codeCommit = new CodeCommitClient({
+      region: region,
+      credentials: credentials,
+    });
+    return codeCommit;
+  }
+
+  async readUrl(
+    url: string,
+    options?: ReadUrlOptions,
+  ): Promise<ReadUrlResponse> {
+    // etag and lastModifiedAfter are not supported by the CodeCommit API
+    try {
+      const { path, repositoryName, region, commitSpecifier } = parseUrl(
+        url,
+        true,
+      );
+      const codeCommitClient = await this.buildCodeCommitClient(
+        this.credsManager,
+        region,
+        this.integration,
+      );
+      const abortController = new AbortController();
+
+      const input: GetFileCommandInput = {
+        repositoryName: repositoryName,
+        commitSpecifier: commitSpecifier,
+        filePath: path,
+      };
+
+      options?.signal?.addEventListener('abort', () => abortController.abort());
+      const getObjectCommand = new GetFileCommand(input);
+      const response: GetFileCommandOutput = await codeCommitClient.send(
+        getObjectCommand,
+        {
+          abortSignal: abortController.signal,
```

Comment:
I believe you should be able to forward `options?.signal` here, just that TypeScript might need a bit of convincing